libsolidity/ast/Types.cpp

```diff
@@ -2808,7 +2818,19 @@ string FunctionType::canonicalName() const
 
 string FunctionType::toString(bool _short) const
 {
-	string name = "function (";
+	string name = "function ";
+	if (m_kind == Kind::Declaration)
+	{
+		// TODO: is this a reasonable way to compute names for these?
+		auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(m_declaration);
+		solAssert(functionDefinition, "");
+		auto const* contract = dynamic_cast<ContractDefinition const*>(functionDefinition->scope());
+		solAssert(contract, "");
+		name += contract->name();
```

Comment:
```suggestion
		name += contract->annotation().canonical_name;
```

Reply:
(at least that is what we do for structs)

Reply:
Yeah - I was looking for something like that - ``fullyQualifiedName`` wasn't nice, but I didn't look further in the annotations. It's safe to assume the annotation will be filled whenever this would be used? I'll check for structs, if it's fine there, it'll be fine here.

Reply:
Yep, should be fine.